aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -752,33 +751,38 @@ Tensor select_sparse_csr(const Tensor& self, int64_t dim, int64_t index) {
 
   // Selecting batch dimension
   if (dim < self.dim() - 2) {
-    if (self.layout() == kSparseBsr) {
```

Comment:
Not sure this is right, now that we have both batch dimensions and dense dimensions supported. cc @pearu

Reply:
Good catch, I have been trying to close off all of these inconsistencies as I go. 

I will need to add tests with dense dims to be sure that it works and is testable, otherwise block it off. Either way the dim check is wrong and I will fix it.